src/Core/Controls/Avatar/AvatarImageSource.cs

```diff
@@ -0,0 +1,62 @@
+﻿using SkiaSharp;
+
+namespace Bit.App.Controls
+{
+    public class AvatarImageSource : StreamImageSource
+    {
+        private readonly string _text;
+        private readonly string _id;
+        private readonly string _color;
+        private readonly AvatarInfo _avatarInfo;
+
+        public override bool Equals(object obj)
+        {
+            if (obj is null)
+            {
+                return false;
+            }
+
+            if (obj is AvatarImageSource avatar)
+            {
+                return avatar._id == _id && avatar._text == _text && avatar._color == _color;
+            }
+
+            return base.Equals(obj);
+        }
+
+        public override int GetHashCode() => _id?.GetHashCode() ?? _text?.GetHashCode() ?? -1;
+
+        public AvatarImageSource(string userId = null, string name = null, string email = null, string color = null)
+        {
+            _id = userId;
+            _text = name;
+            if (string.IsNullOrWhiteSpace(_text))
+            {
+                _text = email;
+            }
+            _color = color;
+
+            //Workaround: [MAUI-Migration] There is currently a bug in MAUI where the actual size of the image is used instead of the size it should occupy in the Toolbar.
+            //This causes some issues with the position of the icon. As a workaround we make the icon smaller until this is fixed.
+            //Github issues: https://github.com/dotnet/maui/issues/12359  and  https://github.com/dotnet/maui/pull/17120
```

Comment:
This issues have been fixed

Reply:
I think @dinisvieira tested it and the issue was still happening. Was it? Maybe we should check again.